File: src/main/java/net/dashflight/data/queue/DashflightRedisQueueModule.java
package net.dashflight.data.queue;

import com.google.inject.AbstractModule;
import com.google.inject.assistedinject.FactoryModuleBuilder;

public class DashflightRedisQueueModule extends AbstractModule {

    @Override
    protected void configure() {
        install(new FactoryModuleBuilder().build(RedisConsumerNodeFactory.class));
    }

}

File: src/main/java/net/dashflight/data/queue/DashflightRedisQueueModule.kt
package net.dashflight.data.queue

import com.google.inject.AbstractModule
import com.google.inject.assistedinject.FactoryModuleBuilder

class DashflightRedisQueueModule : AbstractModule() {
    override fun configure() {
        install(FactoryModuleBuilder().build(RedisConsumerNodeFactory::class.java))
    }
}